### DISCUSSION TITLE
Duplicating children on livewire calls

### DISCUSSION THREAD

**Discussion by MarioPerini** [livewire/livewire#5759] [7. Bugs (archived)]
Copied over from alpine discussion:
https://github.com/alpinejs/alpine/discussions/3647#discussion-5378455

have the problem that elements in a templates are duplicated. I was able to isolate the problem, it only occurs when a fontawesome (svg) element is present.
As soon as alpine calls a @click="$wire.someFunction()", elements are duplicated.
With wire:click="function()" this does not happen. But since I need to pass alpine parameters to livewire, I rely on @click.

I have also tried "x-ignore" and/or "wire:ignore" on the fontawesome element.

Minimal livewire component to reproduce:

Livewire Component (Blade)

```html
<div x-data="test" @click="$wire.double()">
    <template x-for="test in tests" :key="'test_'+test.id">
        <div>
            <div x-text="test.name"></div>
            <template x-for="child in test.child" :key="'child_'+child.id">
                 <div>
                     <div x-text="child.name"></div>
                     <i class="fa-solid fa-file" x-ignore wire:ignore></i>
                  </div>
            </template>
        </div>
    </template>
    <script src="https://kit.fontawesome.com/cdb375a125.js" crossorigin="anonymous"></script>
    <script>
        window.addEventListener("alpine:init", () => {
            Alpine.data("test", () => ({
                tests: [
                    {
                        id: 1,
                        name: 'test1',
                        child: [
                            {
                                id: 1,
                                name: 'test1.1'
                            }
                        ]
                    },
                    {
                        id: 2,
                        name: 'test2',
                        child: [
                            {
                                id: 1,
                                name: 'test2.1'
                            }
                        ]
                    },
                ],
            }));
        });
    </script>
</div>
```

Livewire Component (PHP)
```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Test extends Component
{

    public function double()
    {
        //do something or nothing, doesn't matter
    }

    public function render()
    {
        return view('livewire.test');
    }
}

```

I changed fontawesome temporary from svg to webfont, but in production i have to use svg since i bundle only the used icons for smaller assets.

Am i doing something wrong?

**Comment by MarioPerini** [comment]
Today i could not reproduce this, also happens here:
>  With wire:click="function()" this does not happen.

**Comment by MarioPerini** [comment]
Found out today it doesn't need fontawesome to be present. Also happens if there is an x-cloak in the template.

```html
<div x-data="test" wire:click="double()">
    <template x-for="test in tests" :key="'test_'+test.id">
        <div>
            <div x-text="test.name"></div>
            <template x-for="child in test.child" :key="'child_'+child.id">
                <div x-text="child.name" x-cloak></div>
            </template>
        </div>
    </template>
    <script>
        window.addEventListener("alpine:init", () => {
            Alpine.data("test", () => ({
                tests: [
                    {
                        id: 1,
                        name: 'test1',
                        child: [
                            {
                                id: 1,
                                name: 'test1.1'
                            }
                        ]
                    },
                    {
                        id: 2,
                        name: 'test2',
                        child: [
                            {
                                id: 1,
                                name: 'test2.1'
                            }
                        ]
                    },
                ],
            }));
        });
    </script>
</div>

```

**Comment by atduarte** [comment]
Had a similar issue in Livewire 2. One component was acting weird while others that looked really similar didn't. 

HTML looked good (i.e. it list the items it should) until a Livewire update comes. After that, the existing template items remain there without being tracked by Alpine, while the duplicates do. Tested that by manipulating the _x_data_stack. So something was going on with DOM diffing.

My problem was that I have a hack to forcefully set AlpineJS data on Livewire update (since [@entangled doesn't support dirty](https://github.com/livewire/livewire/discussions/4428)) that was making the DOM diffing go crazy sometimes:
```
<div wire:key="{{ rand() }}" x-cloak>
    <div x-init="hasErrors = {!! json_encode((bool) $hasErrors) !!}"></div>
</div>
```

Livewire 3 seems to have superior DOM diffing and that may solve it for you, but it comes with some other bugs. So my recommendation is to try the techniques described in [here](https://laravel-livewire.com/docs/2.x/troubleshooting#dom-diffing-issues), particularly adding wire:keys to the `template[x-for] > div`.
